CMakeLists.txt

```diff
@@ -1,35 +1,62 @@
+cmake_minimum_required(VERSION 3.25)
 project(bncsutil)
-cmake_minimum_required(VERSION 3.2)
 
 set(CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR} "${PROJECT_SOURCE_DIR}/CMake/Modules")
 message(${CMAKE_BINARY_DIR})
 
 SET(VERSION_MAJOR "1")
 SET(VERSION_MINOR "4")
-SET(VERSION_PATCH "1")
-
-file(GLOB SOURCE
-    "src/bncsutil/*.cpp"
-    "src/bncsutil/*.c"
+SET(VERSION_PATCH "2")
+
+add_library(bncsutil SHARED)
+
+set(HEADERS
+    "src/bncsutil/bncsutil.h"
+    "src/bncsutil/bsha1.h"
+    "src/bncsutil/buffer.h"
+    "src/bncsutil/cdkeydecoder.h"
+    "src/bncsutil/checkrevision.h"
+    "src/bncsutil/decodekey.h"
+    "src/bncsutil/file.h"
+    "src/bncsutil/keytables.h"
+    "src/bncsutil/libinfo.h"
+    "src/bncsutil/ms_stdint.h"
+    "src/bncsutil/mutil.h"
+    "src/bncsutil/mutil_types.h"
+    "src/bncsutil/nls.h"
+    "src/bncsutil/oldauth.h"
+    "src/bncsutil/pe.h"
+    "src/bncsutil/sha1.c"
```

Comment:
The files `sha1.c` and `sha1.h` seem to be specified but only `sha1.h` was supposed to be, it looks like.

Perhaps `sha1.c` should be specified under `SOURCES`?

Reply:
correct, fixed in https://github.com/BNETDocs/bncsutil/pull/18/commits/bc7cc2a78414fa2840e34baec458e17b215d9142